draft-ietf-httpbis-message-signatures.md

```diff
@@ -1555,27 +1617,31 @@ Host: origin.host.internal.example
 Date: Tue, 20 Apr 2021 02:07:56 GMT
 Content-Type: application/json
 Content-Length: 18
-Forwarded: for=192.0.2.123
+Forwarded: for=192.0.2.123;host=example.com;proto=https
 Content-Digest: sha-512=:WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+T\
   aPm+AbwAgBWnrIiYllu7BNNyealdVLvRwEmTHWXvJwew==:
 Signature-Input: sig1=("@method" "@authority" "@path" \
     "content-digest" "content-type" "content-length")\
     ;created=1618884475;keyid="test-key-ecc-p256", \
-  proxy_sig=("signature";key="sig1" "signature-input";key="sig1" \
-    "@authority" "@method" "@path" "forwarded");created=1618884480\
-    ;keyid="test-key-rsa";alg="rsa-v1_5-sha256";expires=1618884540
-Signature: sig1=:e7vqNlt7NV8mJXXja35MQRW/puA3JOHCsKiAG5U7lnpwaqMqn6\
-    83Cna7hyXF4rnpyACdJcOWFB3FSPVymw0fzA==:, \
-  proxy_sig=:V++11u29ZWM1EbQfEx35tnzJVZr8kcoTeuuSCKkdU5J8xr+lb1Ofz7\
-    KuvxxZAkMtTZa4uCdZNxRI+JInYxgc3wUq3duvaugtjmYRe7odUHrEmWKX7m1Ow\
-    R/Q8JoOxSSPcAdtV1knAHXtFCFMdnY3JT8qfnH2yi1VNy65XWCXw4eeK+yxtoE3\
-    DfhtIyzZbmjhcFfXSmibYyJQgIMqagEY/wEsVA2gV+5AE7xFdUrInSGVNVcw86w\
-    9J/stKYC+Deyl3qAzE4qgBN+9VgkuLnUkk6U4W8vk7lSgaq27zUzO9cduybFtRa\
-    E9MBWGaGKEk3Bi+BdIMF06qEjMCnQS+8debQ==:
+  proxy_sig=("@method" "@authority" "@path" "content-digest" \
+    "content-type" "content-length" "forwarded")\
+    ;created=1618884480;keyid="test-key-rsa";alg="rsa-v1_5-sha256"\
+    ;expires=1618884540
+Signature: sig1=:X5spyd6CFnAG5QnDyHfqoSNICd+BUP4LYMz2Q0JXlb//4Ijpzp\
+    +kve2w4NIyqeAuM7jTDX+sNalzA8ESSaHD3A==:, \
+  proxy_sig=:S6ZzPXSdAMOPjN/6KXfXWNO/f7V6cHm7BXYUh3YD/fRad4BCaRZxP+\
+    JH+8XY1I6+8Cy+CM5g92iHgxtRPz+MjniOaYmdkDcnL9cCpXJleXsOckpURl49G\
+    wiyUpZ10KHgOEe11sx3G2gxI8S0jnxQB+Pu68U9vVcasqOWAEObtNKKZd8tSFu7\
+    LB5YAv0RAGhB8tmpv7sFnIm9y+7X5kXQfi8NMaZaA8i2ZHwpBdg7a6CMfwnnrtf\
+    lzvZdXAsD3LH2TwevU+/PBPv0B6NMNk93wUs/vfJvye+YuI87HU38lZHowtznbL\
+    Vdp770I6VHR6WfgS9ddzirrswsE1w5o0LV/g==:
+
 
 {"hello": "world"}
 ~~~
 
+While the proxy could additionally include the client's Signature value and Signature-Input fields from the original message in the new signature's covered components, in this example the proxy does not do this due to known problems in signing signature values as discussed in {{security-sign-signature}}. In some applications, the proxy is a highly trusted component and signing an input signature can be used as a way for the proxy to indicate to the origin server that the proxy has read and verified the client's signature in the original message context. While the origin server may not be able to directly verify this original signature, it can verify that the proxy has vouched for the signature's validity. Use of this approach should be subject to careful analysis of the trust relationship between the different components of the application.
```

Comment:
```suggestion
While the proxy could additionally include the client's Signature value and Signature-Input fields from the original message in the new signature's covered components, in this example the proxy does not do this due to the weaknesses of signing signature values as discussed in {{security-sign-signature}}. If the proxy needs to convey to the origin that it has successfully checked the client's signature, the application SHOULD define a header to indicate so and ensure that the header and all relevant components from the client's request are included within the proxy's signature.    
```

Rationale: I think this is a little clearer on the right path for future adoption, but still compatible with the approach taken by existing applications.